test/common/inc/xdptest.h

```diff
@@ -12,10 +12,14 @@
 #include <winerror.h>
 #include "testframeworkapi.h"
 
+#ifdef KERNEL_MODE
+#define TEST_FAILURE(Format, ...) LogTestFailure(L"" __FILE__, L"" __FUNCTION__, __LINE__, L"" Format, ##__VA_ARGS__)
```

Comment:
IMO we should capture a stack backtrace. It won't render nicely without manually resolving the symbols, but it would sure beat no backtrace at all.

Reply:
RtlCaptureStackBackTrace is the function we'd need

Reply:
You mean something like this? 
```c
#ifdef _KERNEL_MODE
#define TEST_FAILURE(Format, ...) {\
    LogTestFailure(L"" __FILE__, L"" __FUNCTION__, __LINE__, L"" Format, ##__VA_ARGS__); \
    RtlCaptureStackBackTrace();
}
#else
#define TEST_FAILURE(Format, ...) { \
    LogTestFailure(L"" __FILE__, L"" __FUNCTION__, __LINE__, L"" Format, ##__VA_ARGS__); \
    StopTest(); \
}
#endif
```

Reply:
@mtfriesen can you provide a pointer to an example usage?

Reply:
```
    //
    // Capture the caller opening this adapter for diagnostics.
    //
    RtlCaptureStackBackTrace(
        1, RTL_NUMBER_OF(adapterEntry->OpenAdapterStack), adapterEntry->OpenAdapterStack, NULL);
```

Note that the routine can fail, too. Documentation is here: https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/nf-ntifs-rtlcapturestackbacktrace

Reply:
I meant that the LogTestFailure() should be removed or not

Reply:
Using RtlCaptureStackBackTrace get build error. it is hard to fix.

![image](https://github.com/user-attachments/assets/7f4161fe-e344-49f4-9f2b-551e0becebad)

![image](https://github.com/user-attachments/assets/6ff0c86e-f9e5-4ebf-8c6a-d943f3f27b49)

Reply:
Are you including the wpp header? Is WPP processing this file?

Reply:
Ah, yes, this could be tricky to unwrap. I think you might need to create a separate .c file that implements a LogErrorWithBacktrace function and includes its own .tmh file for WPP.

Alternatively, we can use this opportunity to start writing TraceLogging.

Reply:
Any reference?

Reply:
Split PR
https://github.com/microsoft/xdp-for-windows/pull/727